interface.js

```diff
@@ -0,0 +1,75 @@
+$(document).ready(function() {
+  var match;
+
+  new_match();
+
+  $('#score0').on('click', function() { play(0); })
+  $('#score1').on('click', function() { play(1); })
+  $('#score2').on('click', function() { play(2); })
+  $('#score3').on('click', function() { play(3); })
+  $('#score4').on('click', function() { play(4); })
+  $('#score5').on('click', function() { play(5); })
+  $('#score6').on('click', function() { play(6); })
+  $('#score7').on('click', function() { play(7); })
+  $('#score8').on('click', function() { play(8); })
+  $('#score9').on('click', function() { play(9); })
+  $('#score10').on('click', function() { play(10); })
+  $('#newGame').on('click', function() {new_match()})
+
+  function hide_buttons() {
+    for ( i=1; i<=10; i++ ) {
+      if (i <= match.pinsRemaining() || match.currentFrame().isComplete()) {
+        $('#score' + i).fadeIn();
+      } else {
+        $('#score' + i).fadeOut();
+      }
+    }
+  }
+
+  function match_over() {
+    for ( i=0; i<=10; i++ ) { $('#score' + i).hide(); }
+    $('#newGame').show();
+  }
+
+  function clear_scores() {
+    for ( i=1; i<=10; i++) {
+      $('#game' + i + ' #ball1').text('');
+      $('#game' + i + ' #ball2').text('');
+      if ( i == 10 ) { $('#game' + i + ' #ball3').text('')}
+      $('#game' + i + ' #score').text('');
+    }
+    $('#match_score').text('');
+  }
+
+  function new_match() {
+    for ( i=0; i<=10; i++ ) {$('#score' + i).fadeIn();}
+    match = new Match(Frame);
+    $('#newGame').hide();
+    clear_scores();
+  }
+
+  function play(score) {
+    match.play(score);
```

Comment:
So, rather than needing to use indexes in IDs, we could be a little more OO about this by thinking of our UI elements as instances of a class. For example:

```
function FrameView(element, frame) {
  this.frameElement = element;
  this.frame = frame;
}

FrameView.prototype.update() = function() {
  this.frameElement.find('.ball1').text(this.frame.ball1);
  this.frameElement.find('.ball2').text(this.frame.ball2);
  this.frameElement.find('.score').text(this.frame.score());
}

function updateCurrentFrameScores() {
  frameViews.forEach((frame) => { frame.update(); });
}

var frames = $('.frame');
var frameViews = [];

for (var i = 0; i < 10; i++) {
  frameViews.push(new FrameView(frames[i]));
}
```

The nice thing about this approach is that we can then also refactor our view so that it's not repeated:

```
FrameView.prototype.initialiseView() = function() {
  this.frameElement.html('<div class="balls"><span class="ball" id="ball1"></span><span class="ball" id="ball2"></span></div><span class="game_score" id="score"></span>');
}
```
And can remove a lot of the repeated HTML from the index file above.